src/Deserializers/LegacyPropertyDeserializer.php

```diff
@@ -66,32 +72,26 @@ private function getDataTypeId() {
 		return $this->serialization['datatype'];
 	}
 
-	private function setPropertyId() {
-		if ( array_key_exists( 'entity', $this->serialization ) ) {
-			$this->property->setId( $this->getPropertyId() );
-		}
-	}
-
 	/**
-	 * @return PropertyId
 	 * @throws InvalidAttributeException
+	 * @return PropertyId|null
 	 */
 	private function getPropertyId() {
-		$id = $this->idDeserializer->deserialize( $this->serialization['entity'] );
+		if ( array_key_exists( 'entity', $this->serialization ) ) {
```

Comment:
This is going to silently return null if the array key doesn't exist, which is inconsistent with the old behaviour AFAIS.